archunit/src/main/java/com/tngtech/archunit/lang/EvaluationResult.java

```diff
@@ -96,8 +148,8 @@ public Priority getPriority() {
      */
     @PublicAPI(usage = ACCESS)
     public EvaluationResult filterDescriptionsMatching(Predicate<String> linePredicate) {
-        ConditionEvents filtered = new ConditionEvents();
-        for (ConditionEvent event : events) {
+        ConditionEvents filtered = ConditionEvents.Factory.create();
+        for (ConditionEvent event : violations) {
             filtered.add(new FilteredEvent(event, linePredicate));
         }
         return new EvaluationResult(rule, filtered, priority);
```

Comment:
Given that I forgot to transport the `informationAboutNumberOfViolations` I think this is a very good idea :see_no_evil: I adjusted it... But I did not make it public for now.

Reply:
I saw that `informationAboutNumberOfViolations` is not set, but we anyways don't know what to set for the filtered `EvaluationResult`, do we?

Reply:
Yes, I think that's a good idea. It also makes it more explicit that we drop the `informationAboutNumberOfViolations` here (which makes sense I think, since we filter out any number of possible violations, so even if there was an additional hint it's likely not accurate anymore). But I'll keep it out of the public API for now...